contracts/src/v0.8/dev/KeeperRegistryCheckUpkeepGasUsageWrapper.sol

```diff
@@ -0,0 +1,55 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.13;
+
+import "../KeeperRegistry.sol";
+import "../ConfirmedOwner.sol";
+
+/**
+ * @notice This contract serves as a wrapper around a keeper registry's checkUpkeep function.
+ */
+contract KeeperRegistryCheckUpkeepGasUsageWrapper is ConfirmedOwner {
+  KeeperRegistryExecutableInterface private immutable i_keeperRegistry;
+
+  /**
+   * @param keeperRegistry address of a keeper registry
+   */
+  constructor(KeeperRegistryExecutableInterface keeperRegistry) ConfirmedOwner(msg.sender) {
+    i_keeperRegistry = keeperRegistry;
+  }
+
+  /**
+   * @return the keeper registry
+   */
+  function getKeeperRegistry() external view returns (KeeperRegistryExecutableInterface) {
+    return i_keeperRegistry;
+  }
+
+  /**
+   * @notice This function is called by monitoring service to estimate how much gas checkUpkeep functions will consume.
+   * @param id identifier of the upkeep to check
+   * @param from the address to simulate performing the upkeep from
+   */
+  function checkUpkeep(uint256 id, address from)
```

Comment:
I was able to implement this and create a mock for `getState` function.
I was thinking about 3 options to choose a random keeper other than the last keeper to perform `checkUpkeep` function. 
1. always pick the keeper next to the last keeper in the keepers list. (this requires a for loop in sol to look for where the last keeper is in the keeper list) i think this will be expensive and we still need to pick a keeper if the last keeper is not in the keeper list anymore.
2. use some pseudo randomness like block.number/timestamp and pick the (block.number % len(keepers))th keeper in the keepers list. if it's the last keeper, use the next one. This is probably OK bc there is no financial incentive for a miner to manipulate the block number/timestamp.
3. same with the 2 but with VRF (clearly overkill)

i picked the 2nd option. let me know if this makes sense.

an issue here is i want to mock `block.number` in the test. i use this to mine `n`  blocks:
```
for (let index = 0; index < n; index++) {
    await ethers.provider.send('evm_mine', [])
}
```
However, the actual block number i observed in solidity is always `n + 5`... 
For now, i hardcoded `n + 5` in the test as block number. Anyone knows why this is the case? What is the best way to properly mock block.number in this case?